app/controllers/comments_controller.rb

```diff
@@ -240,7 +240,7 @@ def thread_unrestrict
     when 'delete'
       return not_found unless current_user.privilege?('flag_curate') && @comment_thread.deleted?
 
-      if @comment_thread.deleted_by.is_moderator && !current_user.is_moderator
+      if @comment_thread.deleted_by.at_least_moderator? && !current_user.at_least_moderator?
```

Comment:
Untested suggestion:

```suggestion
      if @comment_thread.deleted_by.at_least_moderator? && current_user.standard?
```

Reply:
Same here as with the other checks relying on `current_user` implementation - while it might be safe, let's not invite issues should that change.